parsons/databases/postgres/postgres_core.py

```diff
@@ -0,0 +1,242 @@
+            commit: boolean
+                Whether to commit the transaction immediately. If ``False`` the transaction will
+                be committed when the connection goes out of scope and is closed (or you can
+                commit manually with ``connection.commit()``).
+
+        `Returns:`
+            Parsons Table
+                See :ref:`parsons-table` for output options.
+        """
+
+        # To Do: Have it return an ordered dict to return the
+        #        rows in the correct order
+
+        with self.cursor(connection) as cursor:
+
+            logger.debug(f'SQL Query: {sql}')
+            cursor.execute(sql, parameters)
+
+            if commit:
+                connection.commit()
+
+            # If the cursor is empty, don't cause an error
+            if not cursor.description:
+                logger.debug('Query returned 0 rows')
+                return None
+
+            else:
+
+                # Fetch the data in batches, and "pickle" the rows to a temp file.
+                # (We pickle rather than writing to, say, a CSV, so that we maintain
+                # all the type information for each field.)
+
+                temp_file = files.create_temp_file()
+
+                with open(temp_file, 'wb') as f:
+                    # Grab the header
+                    header = [i[0] for i in cursor.description]
+                    pickle.dump(header, f)
+
+                    while True:
+                        batch = cursor.fetchmany(QUERY_BATCH_SIZE)
+                        if not batch:
+                            break
+
+                        logger.debug(f'Fetched {len(batch)} rows.')
+                        for row in batch:
+                            pickle.dump(list(row), f)
+
+                # Load a Table from the file
+                final_tbl = Table(petl.frompickle(temp_file))
+
+                logger.debug(f'Query returned {final_tbl.num_rows} rows.')
+                return final_tbl
+
+    def _create_table_precheck(self, connection, table_name, if_exists):
+        """
+        Helper to determine what to do when you need a table that may already exist.
+
+        `Args:`
+            connection: obj
+                A connection object obtained from ``redshift.connection()``
+            table_name: str
+                The table to check
+            if_exists: str
+                If the table already exists, either ``fail``, ``append``, ``drop``,
+                or ``truncate`` the table.
+        `Returns:`
+            bool
+                True if the table needs to be created, False otherwise.
+        """
+
+        if if_exists not in ['fail', 'truncate', 'append', 'drop']:
+            raise ValueError("Invalid value for `if_exists` argument")
+
+        exists = self.table_exists_with_connection(table_name, connection)
+
+        if exists and if_exists in ['fail', 'truncate', 'append']:
```

Comment:
Double check me on this, but I think you rewrite as:

```
if exists:
    if exists == 'fail':
        ...
    if exists == 'truncate':
        ...
else:
    if exists == 'drop':
        ...
    return True

return False
```

It simplifies the branching a little bit.

Reply:
Cleaned it up a bit, but there is still a bunch going on unfortunately.